accepted/0000-salt-new-maintaining-workflow.md

```diff
@@ -0,0 +1,252 @@
+
+The `_service` file should look like:
+
+```
+<services>
+  <service name="obs_scm" mode="manual">
+    <param name="url">https://github.com/openSUSE/salt.git</param>
+    <param name="scm">git</param>
+    <param name="versionformat">@PARENT_TAG@</param>
+    <param name="versionrewrite-pattern">v(.*)</param>
+    <param name="revision">openSUSE/devel/master</param>
+    <param name="extract">pkg/suse/salt.*</param>
+  </service>
+  <service name="set_version" mode="manual" />
+  <service name="tar" mode="buildtime"/>
+  <service name="recompress" mode="buildtime">
+    <param name="file">*.tar</param>
+    <param name="compression">gz</param>
+  </service>
+</services>
+``` 
+And it should only contain the following files:
+
+```
+_multibuild
+_service
+salt-xxxx.obscpio
+salt.changes
+salt.obsinfo
+salt.spec
+```
+
+Since services are disabled here, to allow submissions to openSUSE and SLE, this OBS package will be automatically synced with `openSUSE/devel/master` by a Jenkins job.
+
+### OBS and GitHub Webhook integration
+
+As described in the [SCM/CI Workflow integration guide](https://openbuildservice.org/help/manuals/obs-user-guide/cha-obs-scm-ci-workflow-integration#sec-obs-obs-scm-ci-workflow-integration-setup-token-authentication-how-to-authenticate-obs-with-scm), a "GitHub Personal Access Token" must be created and a "GitHub Webhook" configure at `openSUSE/salt` repository.
+
+
+#### OBS workflow file
+
+A `.obs/workflows.yml` will be also added to `openSUSE/salt` to define the OBS workflow as the following:
+
+```
+main_workflow:
+  steps:
+    - branch_package:
+        source_project: systemsmanagement:saltstack:github
+        source_package: salt
+        target_project: systemsmanagement:saltstack:github:CI
+  filters:
+    event: pull_request
+
+rebuild_master:
+  steps:
+    - trigger_services:
+        project: systemsmanagement:saltstack:github
+        package: salt
+  filters:
+    event: push
+    branches:
+      only:
+        - openSUSE/release/xxxx
+```
+
+This workflow will take care of:
+
+- Setting up a new subproject at `systemsmanagement:saltstack:github:CI:...:PR-XXXX/salt` for every incoming PR to build the Salt package according to the changes in the PR.
+- Triggering the services at `systemsmanagement:saltstack:github/salt` on any new push to `openSUSE/release/xxxx` to build the package accordingly.
+
+### Making OBS packages ready to be submitted to Maintenance
+
+Since the package at `systemsmanagement:saltstack:github/salt` has "services" enabled, and we cannot enable/disable services using OBS workflows, this means this package is not yet ready to be submitted to openSUSE or SLE, as they don't accept enabled services on their submissions. We must disable the services.
```

Comment:
Hmmm, interesting. I'll give it a try :+1:

Reply:
I've added some notes about possible usage of SCMSYNC feature from OBS, but after my tests, it does not really work fine when it comes to integrate with the SCM/CI workflow, so the changes on a PR are not actually reflected in the `:github:CI:PR-XXXX` package as seems to be interfering with the SCMSYNC flag.

I gave it a try here:
- OBS package: https://build.opensuse.org/package/show/home:PSuarezHernandez:tests:github/fake-package
- Test PR: https://github.com/meaksh/salt/pull/12
- OBS PR package: https://build.opensuse.org/package/show/home:PSuarezHernandez:tests:github:CI:meaksh:salt:PR-12/fake-package

^^ notice how `salt.changes` on the OBS PR package is not reflecting the actual changes on the PR, and also missing `branch_request` file as we would expect.

I opted for not using `SCMSYNC` and just carrying the `_service` file out of the GitHub repo, which might be actually convenient as differnt OBS/IBS maintained project would has some particularities in its `_service` file, i.a. to point to the right changelog file, so maybe it is easier to just keep them in OBS instead of bringing all those particular files also to Git repo.